typings/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// Type definitions for @callstack/react-theme-provider 1.0.2
+// TypeScript version 3.0.3
+
+declare module "@callstack/react-theme-provider" {
+  import * as React from "react";
+
+  type Without<T, K> = Pick<T, Exclude<keyof T, K>>;
+
+  export type ThemingType<Theme> = {
+    ThemeProvider: typeof ThemeProvider;
+    withTheme: <Props extends { theme: Theme }>(
+      Comp: React.ComponentType<Props>
+    ) => React.ComponentType<Without<Props, "theme">>;
+  };
+
+  export class ThemeProvider<Theme> extends React.Component<{ theme: Theme }> {}
+
+  export const withTheme: <Props extends { theme: {} }>(
+    Comp: React.ComponentType<Props>
+  ) => React.ComponentType<Without<Props, "theme">>;
+
+  export const createTheming: <Theme>(
+    defaultTheme: Theme
+  ) => ThemingType<Theme>;
```

Comment:
In the Flow definitions, `ThemingType` accepts 2 different types, one for the `theme` prop you pass to the provider and receive in the component, and one for the `theme` prop you pass to the component wrapped in `witthTheme(..)`. This is because the `theme` prop for the component doesn't need to be the full theme object, it can be partial, and it'll be deep merged with the active theme before passing it to the underlying component.